core/flyout_nav.js

```diff
@@ -0,0 +1,543 @@
+'use strict';
+
+goog.provide('Blockly.FlyoutNav');
+
+
+
+/*
+*Licensed under the Apache License, Version 2.0 (the "License");
+*you may not use this file except in compliance with the License.
+*You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing, software
+*distributed under the License is distributed on an "AS IS" BASIS,
+*WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*See the License for the specific language governing permissions and
+*limitations under the License.
+*/
+
+/*
+* File overview: this file handles using the keyboard to navigate the flyout
+*/
+
+
+
+/*
+ * class to handle menu navigation
+ * @constructor
+ */
+Blockly.FlyoutNav = function(){
+
+};
+
+/*
+*   Variables for flyout navigation
+*   flyoutArr: everytime the flyout opens the blocks in it are added to this array
+*   oldLengh : size of the array before a new tab opened 
+*   currIndex: index of currently selected block
+*   prevIndex: index of last selected block
+*   opened   : keep track of when the flyout opens and closes 
+*             (needed for selecting blocks when a new category opens)
+*   currentFlyoutArr: array containing all currently shown blocks
+*/
+var flyoutNav = {
+    flyoutArr: [],   
+    oldLength: 0,    
+    currIndex: 0,    
+    prevIndex: 0,
+    opened: false, 
+    currentFlyoutArr: []
+};
+ 
+/*
+* Handles Keyboard input in the toolbox/flyout
+* @param {event}
```

Comment:
It's useful to specify the event type so `@param {KeyboardEvent} e Gen description...`